src/input/reader_to_storage.cc

```diff
@@ -685,6 +811,237 @@ StorageBase * ReaderToStorage::make_transposed_storage(PathBase &p, const Type::
+    try {
+    	ReaderToStorage include_reader(fpath, *(const_cast<Type::Record *>(record)) );
+        return include_reader.get_storage();
+    } catch (ReaderToStorage::ExcInputError &e ) {
+      e << ReaderToStorage::EI_File(fpath); throw;
+    } catch (ReaderToStorage::ExcNotJSONFormat &e) {
+      e << ReaderToStorage::EI_File(fpath); throw;
+    }
+
+	return NULL;
+}
+
+
+StorageBase * ReaderToStorage::make_include_csv_storage(PathBase &p, const Type::Array *array)
+{
+	using namespace boost;
+
+	if ( p.is_record_type() ) { // sub-type must be record type
+		// load path to CSV file
+		std::string included_file;
+        if ( p.down("file") ) {
+       		included_file = get_included_file(p);
+            p.up();
+        } else {
+    	    THROW( ExcInputError() << EI_Specification("Missing key 'file' defines including input file.")
+                               << EI_ErrorAddress(p.as_string()) << EI_InputType(array->desc()) );
+        }
+
+        // number of head lines to skip
+        unsigned int n_head_lines = 0;
+        if ( p.down("n_head_lines") ) {
+        	try {
+        		n_head_lines = p.get_int_value();
+        	}
+			catch (ExcInputError & e) {
+				e << EI_Specification("The value should be '" + p.get_node_type(ValueTypes::int_type) + "', but we found: ");
+				e << EI_ErrorAddress(p.as_string());
+				e << EI_JSON_Type( p.get_node_type(p.get_node_type_index()) );
+				e << EI_InputType("number of lines to skip");
+				throw;
+			}
+        	p.up();
+        }
+
+        // open CSV file, get number of lines, skip head lines
+        FilePath fp((included_file), FilePath::input_file);
+        Tokenizer::Separator separator("\\",",","\"");
+        Tokenizer tok( fp, separator );
+        unsigned int n_lines = 0; // number of lines
+        while ( !tok.eof() ) {
+        	tok.next_line(false);
+        	n_lines++;
+        }
+        if (tok.line().size()==0) n_lines--; // removes last line if it is empty
+        n_lines -= n_head_lines; // subtracts number of skipped head lines
+        tok.set_position( Tokenizer::Position() );
+        for (unsigned int i=0; i<n_head_lines; i++) { // skip head lines
+        	tok.next_line(false);
+        }
+
+        const Type::TypeBase &sub_type = array->get_sub_type(); // sub-type of array
+        StorageBase *item_storage; // storage of sub-type record of included array
+        csv_storage_indexes_.clear();
+        csv_columns_map_.clear();
+        if ( p.down("format") ) {
+			try {
+				item_storage = make_storage(p, &sub_type);
+			} catch (ExcMultipleDefinitionCsvColumn &e) {
+				e << EI_File(tok.f_name());
+				throw;
+			}
+            p.up();
+        } else {
+    	    THROW( ExcInputError() << EI_Specification("Missing key 'format' defines mapping column of CSV file to input subtree.")
+                               << EI_ErrorAddress(p.as_string()) << EI_InputType(array->desc()) );
+        }
+        ASSERT_EQ(csv_storage_indexes_.size(), 0).error();
+
```

Comment:
Get maximal column index from map, check that it is smaller or equal to line length later on during loop over input CSV lines.